include/ginkgo/core/solver/batch_lower_trs.hpp

```diff
@@ -0,0 +1,202 @@
+#include <ginkgo/core/base/lin_op.hpp>
+#include <ginkgo/core/base/types.hpp>
+#include <ginkgo/core/matrix/batch_csr.hpp>
+#include <ginkgo/core/matrix/batch_dense.hpp>
+#include <ginkgo/core/matrix/batch_diagonal.hpp>
+#include <ginkgo/core/matrix/batch_identity.hpp>
+#include <ginkgo/core/matrix/identity.hpp>
+
+namespace gko {
+namespace solver {
+
+
+/**
+ * Solves a batch of linear systems using a batched lower triangular solver.
+ *
+ * @tparam ValueType  precision of matrix elements
+ *
+ * @ingroup solvers
+ * @ingroup BatchLinOp
+ */
+template <typename ValueType = default_precision>
+class BatchLowerTrs : public EnableBatchLinOp<BatchLowerTrs<ValueType>>,
+                      public BatchTransposable {
+    friend class EnableBatchLinOp<BatchLowerTrs>;
+    friend class EnablePolymorphicObject<BatchLowerTrs, BatchLinOp>;
+
+public:
+    using value_type = ValueType;
+    using real_type = gko::remove_complex<ValueType>;
+    using transposed_type = BatchLowerTrs<ValueType>;
+
+    /**
+     * Returns the system operator (matrix) of the linear system.
+     *
+     * @return the system operator (matrix)
+     */
+    std::shared_ptr<const BatchLinOp> get_system_matrix() const
+    {
+        return system_matrix_;
+    }
+
+    /**
+     * @return The left scaling vector.
+     */
+    std::shared_ptr<const BatchLinOp> get_left_scaling_op() const
+    {
+        return left_scaling_;
+    }
+
+    /**
+     * @return The right scaling vector.
+     */
+    std::shared_ptr<const BatchLinOp> get_right_scaling_op() const
+    {
+        return right_scaling_;
+    }
+
+    std::unique_ptr<BatchLinOp> transpose() const override;
+
+    std::unique_ptr<BatchLinOp> conj_transpose() const override;
+
+    GKO_CREATE_FACTORY_PARAMETERS(parameters, Factory)
+    {
+        /**
+         * Batch diagonal matrix for scaling the system matrix from the left
+         * before the solve. Note that `right_scaling_op` must also be set if
+         * this is set.
+         */
+        std::shared_ptr<const BatchLinOp> GKO_FACTORY_PARAMETER_SCALAR(
+            left_scaling_op, nullptr);
+
+        /**
+         * Batch diagonal matrix for scaling the system matrix from the right
+         * before the solve. Note that `left_scaling_op` must also be set if
+         * this is set.
+         */
+        std::shared_ptr<const BatchLinOp> GKO_FACTORY_PARAMETER_SCALAR(
+            right_scaling_op, nullptr);
+
+        /**
+         * @brief Optimization parameter that skips the sorting of the input
+         *        matrix (only skip if it is known that it is already sorted)
+         * (in reference to BatchCsr matrix format).
+         *
+         * The triangular solve algorithm requires the input matrix to be
+         * sorted. If it is, this parameter can be set to `true` to skip the
+         * sorting for better performance.
+         */
+        bool GKO_FACTORY_PARAMETER_SCALAR(skip_sorting, true);
```

Comment:
Is this is good default ? Or maybe we want to set this to `false` by default ?

Reply:
Yeah, it is better to have `false` as the default value here. I made this true by mistake.